earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -96,26 +97,34 @@ void PerformanceLogger::logTiming(
           << duration  << ','
           << size;
 
-  assert(timeFileName.size() > 0);
-  do_notify(message.str(), timeFileName);
+  do_notify(message.str());
 }
 
 // Thread safety wrapper for log output
-void PerformanceLogger::do_notify(const string & message, const string & fileName) {
+void PerformanceLogger::do_notify(const string & message) {
 
   // Get the thread and process IDs
   pthread_t tid = pthread_self();
   pid_t pid = getpid();
 
   {  // atomic inner block
     plLockGuard lock( write_mutex );
-    { // Make sure we flush and close the output file before unlocking the mutex:
-      ofstream output_stream(fileName.c_str(), ios::app);
-      output_stream << pid << ',' << tid << ',' << message << endl;
+    {
+      if (buffer.size() + message.size() > BUF_SIZE) {
```

Comment:
@tst-lsavoie: Have you compared performance when using an external `std::string` as a buffer vs. setting `std::ofstream`'s `streambuf`?  (E.g. `timeFile.rdbuf()->pubsetbuf(buffer.c_str(), buffer.capacity());`)

Reply:
Yes, I did try that. It didn't seem to have an effect. Apparently `pubsetbuf` is not guaranteed to do anything.